src/sentry/middleware/integrations/parsers/slack.py

```diff
@@ -126,10 +128,17 @@ def get_response(self):
         if data and is_event_challenge(data):
             return self.get_response_from_control_silo()
 
-        regions = self.get_regions_from_organizations()
-        if len(regions) == 0:
-            logger.info("%s.no_regions", self.provider, extra={"path": self.request.path})
-            return self.get_response_from_control_silo()
+        try:
+            regions = self.get_regions_from_organizations()
+        except Integration.DoesNotExist:
+            # Alert, as there may be a misconfiguration issue
+            sentry_sdk.capture_exception()
+            return self.get_default_missing_integration_response()
+        except OrganizationIntegration.DoesNotExist:
+            # Swallow this exception, as this is likely due to a user removing
+            # their org's slack integration, and slack will continue to retry
+            # this request until it succeeds.
+            return HttpResponse(status=202)
```

Comment:
Should we add logging or a metric here, so we understand the amount of dead-installation webhooks we're receiving?

Reply:
Nevermind, I see it now as part of the base parser -- good call.